tests/test_lineage.py

```diff
@@ -495,3 +495,82 @@ def test_node_name_doesnt_contain_comment(self) -> None:
         self.assertEqual(len(node.downstream), 1)
         self.assertEqual(len(node.downstream[0].downstream), 1)
         self.assertEqual(node.downstream[0].downstream[0].name, "t1.x")
+
+    def test_pivot_without_alias(self) -> None:
+        sql = """
+        SELECT 
+            a as other_a
+        FROM (select value,category from sample_data)
+        PIVOT (
+            sum(value)
+            FOR category IN ('a', 'b')
+        );
+        """
+        node = lineage("other_a", sql)
+
+        self.assertEqual(node.downstream[0].name, "_q_0.value")
+        self.assertEqual(node.downstream[0].downstream[0].name, "sample_data.value")
```

Comment:
Is this correct? Shouldn't the first node downstream be the `PIVOT` source? I.e. if the pivot ends up being aliased to `_q_1`, you'd say that this name is `_q_1.a`, and then go to `_q_0.value` and finally to `sample_data.value`.

Reply:
Ditto for the remaining tests as well.

Reply:
Not sure if I understand that correctly.

If I check how this sql is qualified:
```sql
SELECT 
    a as other_a
FROM (select value,category from sample_data)
PIVOT (
    sum(value)
    FOR category IN ('a', 'b')
);
```
to
```sql
SELECT 
    _q_1.a AS other_a 
FROM 
    (
        SELECT 
            sample_data.value AS value, 
            sample_data.category AS category 
        FROM 
            sample_data AS sample_data
    ) AS _q_0 
PIVOT (
    SUM(_q_0.value) 
    FOR _q_0.category IN ('a', 'b')
) AS _q_1
```

`_q_0` is the alias of the subquery and the pivot aggregation `SUM(_q_0.value)` uses the column `_q_0.value` which I would expect in the first downstream node.

Do I misunderstand something here?

Reply:
Yeah, my suggestion was to produce the following lineage:

```
             +~~~ this is missing right now
             v
other_a -> _q_1.a -> _q_0.value -> sample_data.value
```

I think we do something similar in this simpler example:

```
>>> node = lineage("a", "select _q_0.a as a from (select t.col as a from t) _q_0")
>>> node.downstream
[Node(name='_q_0.a', expression=Alias(
  this=Column(
    this=Identifier(this=col, quoted=False),
    table=Identifier(this=t, quoted=False)),
  alias=Identifier(this=a, quoted=False)), source=Select(
  expressions=[
    Alias(
      this=Column(
        this=Identifier(this=col, quoted=False),
        table=Identifier(this=t, quoted=False)),
      alias=Identifier(this=a, quoted=False))],
  from=From(
    this=Table(
      this=Identifier(this=t, quoted=False),
      alias=TableAlias(
        this=Identifier(this=t, quoted=False))))), downstream=[Node(name='t.col', expression=Table(
  this=Identifier(this=t, quoted=False),
  alias=TableAlias(
    this=Identifier(this=t, quoted=False))), source=Table(
  this=Identifier(this=t, quoted=False),
  alias=TableAlias(
    this=Identifier(this=t, quoted=False))), downstream=[], source_name='', reference_node_name='')], source_name='', reference_node_name='_q_0')]
```

Notice how the first node downstream is `_q_0.a` .

Reply:
Now I understand what you mean.

In your example the lineage node name and source are:
```
   a       / select _q_0.a as a from (select t.col as a from t) _q_0
-> _q_0.a  / select t.col as a from t
-> t.col   / t AS t
```
So `_q_0` is a more specific scope.

The pivot example:
```
   other_a           / SELECT _q_1.a AS other_a FROM (SELECT sample_data.value AS value, sample_data.category AS category FROM sample_data AS sample_data) AS _q_0 PIVOT(SUM(_q_0.value) FOR _q_0.category IN ('a', 'b')) AS _q_1
-> _q_0.value        / SELECT sample_data.value AS value FROM sample_data AS sample_data
-> sample_date.value / sample_data AS sample_data
```
Here `_q_1` it not a more specific scope, because the pivot is not a scope itself. I can't select a proper more specific source for this step.
If I use the whole scope again I would get:
```
   other_a           / SELECT _q_1.a AS other_a FROM (SELECT sample_data.value AS value, sample_data.category AS category FROM sample_data AS sample_data) AS _q_0 PIVOT(SUM(_q_0.value) FOR _q_0.category IN ('a', 'b')) AS _q_1
-> _q_1.a            / SELECT _q_1.a AS other_a FROM (SELECT sample_data.value AS value, sample_data.category AS category FROM sample_data AS sample_data) AS _q_0 PIVOT(SUM(_q_0.value) FOR _q_0.category IN ('a', 'b')) AS _q_1
-> _q_0.value        / SELECT sample_data.value AS value FROM sample_data AS sample_data
-> sample_date.value / sample_data AS sample_data
```
But because the source is identical in both steps, that would end in an endless loop.

So I'm not sure how to continue from here

Reply:
I see, that's fine then. If we do `other_a -> _q_0.value -> sample_data.value`, is the `PIVOT` operator present anywhere in the lineage tree? I'd like to be able to display that there's a pivot happening that leads to this column. If that's true, then we can leave it as is I think.